tools/bin/ecflow/def_files/make_suite_def.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python
+"""
+make_suite_def.py
+"""
+import os
+import os.path
+import sys
+import argparse
+
+import ecflow
+
+from tonic.io import read_config
+from monitor.log import set_logger
+
+# --------------------------------------------------------------------------- #
+def main():
+    """
+    Write suite definition file for over-the-loop data processing.
+    """
+
+    logger = set_logger(os.path.splitext(os.path.split(__file__)[-1])[0],
+                        'INFO')
+    parser = argparse.ArgumentParser(description='Create suite definition file')
+    parser.add_argument('suite_config_file', metavar='suite_config_file',
+                        type=argparse.FileType('r'), nargs=1,
+                        help='configuration file for suite definition file')
+    args = parser.parse_args()
+    suite_dict = read_config(args.suite_config_file[0].name)
+    logger.info('Creating suite definition')
+    defs = ecflow.Defs()
+
+    for s in range(suite_dict['MAIN']['NSuites']):
+        skey = 'S{}'.format(s)
+        suite = defs.add_suite(suite_dict[skey]['Name'])
+        suite = add_dependencies(suite, skey, suite_dict)
+        suite.add_variable('SYS_PATH', os.environ['PATH'])
+
+        for f in range(suite_dict[skey]['NFamilies']):
+            fkey = 'S{0}_F{1}'.format(s, f)
+            family = suite.add_family(suite_dict[fkey]['Name'])
+            family = add_dependencies(family, fkey, suite_dict)
+            for t in range(suite_dict[fkey]['NTasks']):
+                tkey = 'S{0}_F{1}_T{2}'.format(s, f, t)
+                task = family.add_task(suite_dict[tkey]['Name'])
+                task = add_dependencies(task, tkey, suite_dict)
+            for ff in range(suite_dict[fkey]['NFamilies']):
+                ffkey = 'S{0}_F{1}_F{2}'.format(s, f, ff)
+                ffamily = family.add_family(suite_dict[ffkey]['Name'])
+                ffamily = add_dependencies(ffamily, ffkey, suite_dict)
+                for t in range(suite_dict[ffkey]['NTasks']):
+                    tkey = 'S{0}_F{1}_F{2}_T{3}'.format(s, f, ff, t)
+                    task = ffamily.add_task(suite_dict[tkey]['Name'])
+                    task = add_dependencies(task, tkey, suite_dict)
+
+    logger.info(defs)
+    if suite_dict['MAIN']['MakeDummyScripts'] == True:
```

Comment:
you don't need the ` == True` part here.  If you want to differentiate between `TRUE` and `1` (for example).  You need `if x is True:`